### PR TITLE
Replacement Tables REST API list of methods added

### PR DESCRIPTION
**Pull request by raipc**

### CODE REVIEW

**Comment by raipc** [comment]
The documentation leaks examples and description of methods to work with CSV.

**Comment by rodionos** [comment]
The current base examples are sufficient.
@raipc - Please add new API docs to TOC and config.js after the **Entity Group** section.
Assign to @kghamilton89 for review when ready.
Keep our new [guide](https://github.com/axibase/docs-util/blob/master/guide.md) in mind when reviewing the docs.

**Comment by rodionos** [comment]
@kghamilton89 - pls review

File: api/meta/replacement-table/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Meta API: Replacement Tables Methods
+
+| **Name** | **Method** | **Path** | **Accept** | **Content-Type** | **Description** |
+|:---|:---|:---|:---|:---|:----|
+| [Get](get.md) | GET | `/api/v1/replacement-tables/csv/{name}` | | `text/csv` | Retrieve replacement table keys and values in CSV format and metadata in link header. |
```
Comment (rodionos) [inline api/meta/replacement-table/README.md]: `Link` should be backticked. It's the name of the header. See: https://github.com/axibase/docs-util/blob/master/guide.md#backticks - header names. Also link to Link header section in get.md

**Comment by kghamilton89** [comment]
@raipc Travis build is failing based on the inclusion of a blacklisted work ([admin](https://travis-ci.org/axibase/atsd/builds/393985892#L730)) @ line 103 of toc.md, which you'll see doesn't have the word 'admin' but 'administrator.'

File: api/meta/replacement-table/create-or-replace.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+| **Name** | **Description** |
+|:---|:---|
+| `format` | **[Required]** Output format: `json` or `csv`. |
+| `name` | **[Required]** Replacement table name. |
+
+### Fields
+
+If `format` is `json`, refer to Response Fields in [Replacement Table: Get](get.md#fields).
+
+If `format` is `csv`, the request body must contain records in CSV format without header.
```
Comment (rodionos) [inline api/meta/replacement-table/create-or-replace.md]: Here, I'd like to highlight the fact that is a fully correct way of backticking. While CSV is the format, the `cvs` is the name of the parameter and therefore is backticked. Good job.

File: api/meta/replacement-table/create-or-replace.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+#### curl
+
+```bash
+curl https://atsd_hostname:8443/api/v1/replacement-tables/csv/pi_pids \
+  --insecure --include --user {username}:{password} \
+  --request PUT \
+  --header 'Content-Type: text/csv' \
+  --data @- <<'EOF'
```
Comment (rodionos) [inline api/meta/replacement-table/create-or-replace.md]: We don't typically use this way of sending curl payload. I prefer to keep things consistent. It's a multiline content, so perhaps we should display how the content is sent from file, e.g. `pids.csv`?

File: api/meta/replacement-table/get.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
+# Replacement Table: Get
+
+## Description
+
+Retrieves records and metadata for the specified replacement table.
+
+If `format` is `json`, the response contains a single object with metadata and records.
+
+If `format` is `csv`, the response contains records in CSV format, and metadata is specified as JSON Linked Data (JSON-LD) in Link header.
```
Comment (rodionos) [inline api/meta/replacement-table/get.md]: Backtick the `Link` which is the name of the header. Add link to [W3C Model for Tabular Data](https://www.w3.org/TR/tabular-data-model/)

File: api/meta/replacement-table/get.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+| **Name** | **Description** |
+|:---|:---|
+| `format` | **[Required]** Output format: `json` or `csv`. |
+| `name` | **[Required]** Replacement table name. |
+
+### Query String Parameters
+
+| **Name** | **Type** | **Description** |
+|:---|:---|:---|
+| `comments` | boolean | Boolean statement which determines whether response includes comments in output. Works with `csv` format only. |
```
Comment (rodionos) [inline api/meta/replacement-table/get.md]: > Boolean statement which determines whether response includes comments in output. Works with `csv` format only.

Not really a statement. It's a field or parameter. For consistency, use phrasing that we use for other boolean fields in other API examples. `addComments` would have been better (for future reference).

File: api/meta/replacement-table/get.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+### Fields
+
+If `format` is `json`, the response includes a JSON object which includes the following values:
+
+|**Name**|**Type**|**Description**|
+|:---|:---|:---|
+| `name` |string|Replacement table name.|
+| `author` |string|Replacement table creator.|
+| `description` |string|Replacement table description.|
+| `format` |string|Replacement table UI representation.|
```
Comment (rodionos) [inline api/meta/replacement-table/get.md]: What is 'UI representation'? Needs to be clarified.

File: api/meta/replacement-table/update.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+## Description
+
+Updates records and metadata of the specified replacement table.
+
+Unlike the [replace method](create-or-replace.md), records and metadata that are **not** specified by the request remain unchanged.
+
+Similarly, fields set to `null` remain unchanged.
+
+If content type is JSON, request updates both metadata and records. If content type is CSV, request updates only records.
```
Comment (rodionos) [inline api/meta/replacement-table/update.md]: In this case, the content type should be lower case and backticked? These are specific, case-sensitive values.

File: api/meta/replacement-table/README.md (hunk 1 above)
Comment (kghamilton89) [reply api/meta/replacement-table/README.md]: Replaced this and other instances.

File: api/meta/replacement-table/create-or-replace.md (hunk 2 above)
Comment (kghamilton89) [reply api/meta/replacement-table/create-or-replace.md]: :+1:

File: api/meta/replacement-table/get.md (hunk 4 above)
Comment (kghamilton89) [reply api/meta/replacement-table/get.md]: This is linked later in the document, but I've moved it to the first mention.

File: api/meta/replacement-table/get.md (hunk 5 above)
Comment (kghamilton89) [reply api/meta/replacement-table/get.md]: Changed to parameter.

File: api/meta/replacement-table/update.md (hunk 7 above)
Comment (kghamilton89) [reply api/meta/replacement-table/update.md]: replaced.

File: api/meta/replacement-table/create-or-replace.md (hunk 3 above)
Comment (kghamilton89) [reply api/meta/replacement-table/create-or-replace.md]: @raipc

File: api/meta/replacement-table/get.md (hunk 6 above)
Comment (kghamilton89) [reply api/meta/replacement-table/get.md]: @raipc

**Comment by raipc** [comment]
What's wrong with `toc.md`? I haven't found **Admin** there on line 103.

**Comment by kghamilton89** [comment]
@raipc Yes that's my point. Travis build fails and the only error it cites is the presence of blacklisted word in `toc.md`. There is a problem with Travis.

**Comment by raipc** [comment]
Created a [PR](https://github.com/axibase/docs-util/pull/4)

File: api/meta/replacement-table/get.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+| **Name** | **Description** |
+|:---|:---|
+| `format` | **[Required]** Output format: `json` or `csv`. |
+| `name` | **[Required]** Replacement table name. |
+
+### Query String Parameters
+
+| **Name** | **Type** | **Description** |
+|:---|:---|:---|
+| `comments` | boolean | Boolean parameter which determines whether response includes comments in output. Works with `csv` format only. |
```
Comment (rodionos) [inline api/meta/replacement-table/get.md]: `works ...` is informal, replace.

Comment (rodionos) [inline api/meta/replacement-table/get.md]: It's already clear from type that the parameter is boolean.

File: api/meta/replacement-table/get.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+### Fields
+
+If `format` is `json`, the response includes a JSON object which includes the following values:
+
+|**Name**|**Type**|**Description**|
+|:---|:---|:---|
+| `name` |string|Replacement table name.|
+| `author` |string|Replacement table creator.|
+| `description` |string|Replacement table description.|
+| `format` |string| Format used for syntax highlighting on **Data > Replacement Tables > {name}** form |
```
Comment (rodionos) [inline api/meta/replacement-table/get.md]: What `format` values are supported? I don't like how this parameter overlaps with format (csv | json). Source of collision going forward.

File: api/meta/replacement-table/create-or-replace.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
+# Replacement Table: Create or Replace
+
+## Description
+
+Creates a replacement table with specified name, description, type, author, and records.
+
+If the specified replacement table exists, all current replacement table records and metadata are replaced by those specified in the request.
```
Comment (rodionos) [inline api/meta/replacement-table/create-or-replace.md]: What does it mean `all current replacement table records`? Please clarify with dev what will be replace.

**Comment by raipc** [comment]
@rodionos What about `syntax`? Should the field be renamed in xml-serialized form?

**Comment by rodionos** [comment]
`syntax` is ok too, another option is `valueSyntax` because this field controls what language we set in codemirror to display values in the replacement table. `valueFormat` is another option -  it keep the original field name but clarifies it to avoid collision.

**Comment by raipc** [comment]
Thanks, I like `valueFormat`

File: api/meta/replacement-table/get.md (hunk 8 above)
Comment (kghamilton89) [reply api/meta/replacement-table/get.md]: Applies to...

Comment (kghamilton89) [reply api/meta/replacement-table/get.md]: removed second boolean

File: api/meta/replacement-table/get.md (hunk 9 above)
Comment (kghamilton89) [reply api/meta/replacement-table/get.md]: Body parameter name updated and parameter options included.

File: api/meta/replacement-table/create-or-replace.md (hunk 10 above)
Comment (kghamilton89) [reply api/meta/replacement-table/create-or-replace.md]: rephrased for clarity.

**Pull request merged**